tests/test_c_code/CMakeLists.txt

```diff
@@ -0,0 +1,91 @@
+cmake_minimum_required(VERSION 3.15)
+project(cantoolsTest C)
+set(CMAKE_C_STANDARD 99)
+
+# Regenerate C files
+execute_process(COMMAND cantools generate_c_source ../files/dbc/motohawk.dbc 
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+execute_process(COMMAND cantools generate_c_source ../files/dbc/padding_bit_order.dbc 
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+execute_process(COMMAND cantools generate_c_source ../files/dbc/vehicle.dbc 
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+execute_process(COMMAND cantools generate_c_source ../files/dbc/open_actuator.dbc 
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+execute_process(COMMAND cantools generate_c_source ../files/dbc/multiplex.dbc 
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+execute_process(COMMAND cantools generate_c_source ../files/dbc/multiplex_2.dbc 
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+execute_process(COMMAND cantools generate_c_source ../files/dbc/floating_point.dbc 
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+execute_process(COMMAND cantools generate_c_source ../files/dbc/no_signals.dbc 
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+execute_process(COMMAND cantools generate_c_source ../files/dbc/choices.dbc 
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+execute_process(COMMAND cantools generate_c_source ../files/dbc/signed.dbc 
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+execute_process(COMMAND cantools generate_c_source ../files/sym/min-max-only-6.0.sym 
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+execute_process(COMMAND cantools generate_c_source ../files/dbc/abs.dbc 
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+execute_process(COMMAND cantools generate_c_source --database-name my_database_name ../files/dbc/motohawk.dbc 
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+execute_process(COMMAND cantools generate_c_source --database-name motohawk_bit_fields --bit-fields ../files/dbc/motohawk.dbc 
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+execute_process(COMMAND cantools generate_c_source --database-name floating_point_bit_fields --bit-fields ../files/dbc/floating_point.dbc 
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+execute_process(COMMAND cantools generate_c_source --database-name signed_bit_fields --bit-fields ../files/dbc/signed.dbc 
+                WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
+
+
+# fetch unity test framework
+include(FetchContent)
+FetchContent_Declare(
+    unity
+    GIT_REPOSITORY https://github.com/ThrowTheSwitch/Unity.git
+    GIT_TAG v2.6.0
+)
+FetchContent_MakeAvailable(unity)
```

Comment:
hm, I'm not sure if it is a good idea to (potentially) retrieve stuff from the internet as part of the `cmake` config process. IMO, the clean solution would be to install the dependency before and `Find` it using cmake. That said, given that this might become quite iffy and the unity testing framework is comprised of only three files, I think that simply grabbing the latest copy and shipping it as part of cantools is a valid option as well...

Reply:
I thought about this and i'm not sure if the other options are really better. I've found a few example projects which use git submodules for Unity. That would work but would require the user to clone recursively.

Reply:
why not vendor it? it is set to the v2.6.0 tag anyway and I think we should not assume that the machine which runs the test has access to the internet at configure time...

Reply:
Done :)